src/eth/rpc/rpc_parser.rs

```diff
@@ -54,8 +60,28 @@ pub fn rpc_parsing_error<S: serde::Serialize>(message: S) -> ErrorObjectOwned {
     ErrorObjectOwned::owned(PARSE_ERROR_CODE, PARSE_ERROR_MSG, Some(message))
 }
 
-impl From<EthError> for ErrorObjectOwned {
-    fn from(error: EthError) -> Self {
-        rpc_internal_error(error.to_string())
+pub struct RPCError(anyhow::Error);
+impl Display for RPCError {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        Display::fmt(&self.0, f)
+    }
+}
+impl Debug for RPCError {
```

Comment:
Can you just derive `Debug` instead?

Reply:
Well, kinda. If I derive Debug formatting `RPCError` will produce something like this: `"RPCError(inner.fmt())"`, but what I wanted was for RPCError to behave exactly like an anyhow::Error (although I didn't implement Deref, because I wasn't sure we'd need it and I did not want to bloat this struct). I can change it If you think it'd be better.

Reply:
For `Display` trait, it is fine to delegate to the inner implementation because the goal is an human-friendly message. But for `Debug`, we should stick with the struct representation when possible.

It is the same idea of `repr` and `str` from Python.